tests/smoke/smoke004-uses-events.html

```diff
@@ -0,0 +1,50 @@
+<html>
+    <head>
+        <title>Injectorjs - Injector Complete Event</title>
+
+        <script>
+            var injectorReady = false;
+            var domReady = false;
+
+            function domLoaded () {
+              domReady = true;
+              loadedHandler();
+            }
+
+            function injectorLoaded () {
+              injectorReady = true;
+              loadedHandler();
+            }
+
+            function loadedHandler () {
+              if (injectorReady && domReady) {
+                document.body.innerHTML = "<div id='injector-success'>Injector Loaded Event And Document Loaded Event Used</div>";
+              }
+            }
+
+
+            if( typeof FAI === 'undefined' || FAI === null ){
+              document.addEventListener("DOMContentLoaded", injectorLoaded);
+            } else {
+              if (FAI !== null && FAI.INJECTOR && FAI.INJECTOR.readyState) {
+                injectorReady = true;
+              } else {
+                document.addEventListener("DOMContentLoaded", injectorLoaded);
```

Comment:
also, when you call injectorLoaded here when the "DOMContentLoaded" is thrown, you set injectorReady to true, but the script for the injector is being loaded async below the body tag so just because the body tag is loaded and the "DOMContentLoaded" event is thrown doesn't mean that the injector has fully loaded.